torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1276,13 +1277,46 @@ ProcessGroupNCCL::~ProcessGroupNCCL() {
 #endif
 }
 
-void dump_debugging_info() {
+void storeTraceInfoToLocalDisk(int rank, const std::string& ncclTrace) {
```

Comment:
i'm wondering if it is cleaner to define a 'class' for the disk dumper, so it can have a constructor that initializes things like its filename, and avoid processing envs in the actual write to disk method.

and then we'd also change the interface for registering from function to object following that interface.

i dont have a strong feeling here, maybe ok to just leave as function.

Reply:
I see. Yeah, making it a class would be better and later on users can just override the class and do their customized logic there.

Reply:
Let's also put this in `TestUtils.hpp`, WDYT?

Reply:
@XilunWu this is the default fallback function we will need to keep in this file. This is not for testing purpose at all.